explorer/src/components/account/history/TokenTransfersDetails.tsx

```diff
@@ -0,0 +1,213 @@
+  TokenInstructionType,
+  Transfer,
+  TransferChecked,
+} from "components/instruction/token/types";
+import { InstructionContainer } from "utils/instruction";
+import { Signature } from "components/common/Signature";
+import { Address } from "components/common/Address";
+import { normalizeTokenAmount } from "utils";
+import Moment from "react-moment";
+
+type MintDetails = {
+  decimals: number;
+  mint: string;
+};
+
+export function TransfersDetails({
+  pubkey,
+  slotRows,
+}: {
+  pubkey: PublicKey;
+  slotRows: SlotRow[];
+}) {
+  const address = pubkey.toBase58();
+  const history = useAccountHistory(address);
+  const fetchDetailedAccountHistory = useFetchDetailedAccountHistory(pubkey);
+  const detailedHistoryMap = useDetailedAccountHistory(address);
+  const { tokenRegistry } = useTokenRegistry();
+  const mintDetails = tokenRegistry.get(pubkey.toBase58());
+
+  React.useEffect(() => {
+    if (history?.data?.fetched) {
+      fetchDetailedAccountHistory(history.data.fetched);
+    }
+  }, [history]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const hasTimestamps = !!slotRows.find((element) => !!element.blockTime);
+  const detailsList: React.ReactNode[] = [];
+  const mintMap = new Map<string, MintDetails>();
+
+  slotRows.forEach(({ signature, blockTime, statusText, statusClass }) => {
+    const parsed = detailedHistoryMap.get(signature);
+    if (!parsed) return;
+
+    // Extract mint information from token deltas
+    // (used to filter out non-checked tokens transfers not belonging to this mint)
+    if (parsed.meta?.preTokenBalances) {
+      parsed.meta.preTokenBalances.forEach((balance) => {
+        const account =
+          parsed.transaction.message.accountKeys[balance.accountIndex];
+        mintMap.set(account.pubkey.toBase58(), {
+          decimals: balance.uiTokenAmount.decimals,
+          mint: balance.mint,
+        });
+      });
+    }
+
+    if (parsed.meta?.postTokenBalances) {
+      parsed.meta.postTokenBalances.forEach((balance) => {
+        const account =
+          parsed.transaction.message.accountKeys[balance.accountIndex];
+        mintMap.set(account.pubkey.toBase58(), {
+          decimals: balance.uiTokenAmount.decimals,
+          mint: balance.mint,
+        });
+      });
+    }
+
+    // Extract all transfers from transaction
+    let transfers: (Transfer | TransferChecked)[] = [];
+    InstructionContainer.create(parsed).instructions.forEach(
+      ({ instruction, inner }) => {
+        const transfer = getTransfer(instruction);
+        if (transfer) {
+          transfers.push(transfer);
+        }
+        inner.forEach((instruction) => {
+          const transfer = getTransfer(instruction);
+          if (transfer) {
+            transfers.push(transfer);
+          }
+        });
+      }
+    );
+
+    // Filter out transfers not belonging to this mint
+    transfers = transfers.filter((transfer) => {
+      if ("tokenAmount" in transfer && transfer.mint !== pubkey) {
```

Comment:
```suggestion
      if ("tokenAmount" in transfer && !transfer.mint.equals(pubkey)) {
```
This check is excluding all checked transfers